great_expectations/expectations/conditions.py

```diff
@@ -107,22 +130,12 @@ def dict(self, **kwargs) -> dict:
         return result
 
     def __and__(self, other: Condition) -> AndCondition:
-        new_conditions = []
-        for cond in [self, other]:
-            if isinstance(cond, AndCondition):
-                new_conditions.extend(cond.conditions)
-            else:
-                new_conditions.append(cond)
-        return AndCondition(conditions=new_conditions)
+        """Construct an AndCondition"""
+        return AndCondition(conditions=[self, other])
 
```

Comment:
This flattening logic was moved into the `row_condition` validator. Flattening needs to happen there in case somebody decides to pass `AndCondition`s directly (as opposed to using the & operator).